src/ol-ext/format/featurehash.js

```diff
@@ -0,0 +1,940 @@
+ * @param {Array.<string>} encodedStyles Array of encoded strings.
+ * @private
+ */
+ngeo.format.FeatureHash.encodeStylePolygon_ =
+    function(fillStyle, strokeStyle, encodedStyles) {
+  ngeo.format.FeatureHash.encodeStyleFill_(fillStyle, encodedStyles);
+  if (!goog.isNull(strokeStyle)) {
+    ngeo.format.FeatureHash.encodeStyleStroke_(strokeStyle, encodedStyles);
+  }
+};
+
+
+/**
+ * @param {ol.style.Fill} fillStyle Fill style.
+ * @param {Array.<string>} encodedStyles Array of encoded strings.
+ * @param {string=} opt_propertyName Property name.
+ * @private
+ */
+ngeo.format.FeatureHash.encodeStyleFill_ =
+    function(fillStyle, encodedStyles, opt_propertyName) {
+  var propertyName = goog.isDef(opt_propertyName) ?
+      opt_propertyName : 'fillColor';
+  var fillColor = fillStyle.getColor();
+  if (!goog.isNull(fillColor)) {
+    var fillColorRgba = ol.color.asArray(fillColor);
+    var fillColorHex = goog.color.rgbArrayToHex(fillColorRgba);
+    if (encodedStyles.length > 0) {
+      encodedStyles.push('\'');
+    }
+    encodedStyles.push(
+        encodeURIComponent(propertyName + '*' + fillColorHex));
+  }
+};
+
+
+/**
+ * @param {ol.style.Stroke} strokeStyle Stroke style.
+ * @param {Array.<string>} encodedStyles Array of encoded strings.
+ * @private
+ */
+ngeo.format.FeatureHash.encodeStyleStroke_ =
+    function(strokeStyle, encodedStyles) {
+  var strokeColor = strokeStyle.getColor();
+  if (!goog.isNull(strokeColor)) {
+    var strokeColorRgba = ol.color.asArray(strokeColor);
+    var strokeColorHex = goog.color.rgbArrayToHex(strokeColorRgba);
+    if (encodedStyles.length > 0) {
+      encodedStyles.push('\'');
+    }
+    encodedStyles.push(encodeURIComponent('strokeColor*' + strokeColorHex));
+  }
+  var strokeWidth = strokeStyle.getWidth();
+  if (goog.isDef(strokeWidth)) {
+    if (encodedStyles.length > 0) {
+      encodedStyles.push('\'');
+    }
+    encodedStyles.push(encodeURIComponent('strokeWidth*' + strokeWidth));
+  }
+};
+
+
+/**
+ * @param {ol.style.Text} textStyle Text style.
+ * @param {Array.<string>} encodedStyles Array of encoded strings.
+ * @private
+ */
+ngeo.format.FeatureHash.encodeStyleText_ = function(textStyle, encodedStyles) {
+  var fontStyle = textStyle.getFont();
+  if (goog.isDef(fontStyle)) {
+    var font = fontStyle.split(' ');
+    if (font.length >= 3) {
+      if (encodedStyles.length > 0) {
+        encodedStyles.push('\'');
+      }
+      encodedStyles.push(encodeURIComponent('fontSize*' + font[1]));
+    }
+  }
+  var fillStyle = textStyle.getFill();
+  if (!goog.isNull(fillStyle)) {
+    ngeo.format.FeatureHash.encodeStyleFill_(
+        fillStyle, encodedStyles, 'fontColor');
+  }
+};
+
+
+/**
+ * @param {string} text Text.
+ * @param {olx.format.ReadOptions=} opt_options Read options.
+ * @return {ol.geom.LineString} Line string.
+ * @this {ngeo.format.FeatureHash}
+ * @private
+ */
+ngeo.format.FeatureHash.readLineStringGeometry_ = function(text, opt_options) {
```

Comment:
True. It's related to a limitation, where the format doesn't support transforming the decoded geometries. I'll remove the unused `opt_options`.